### DISCUSSION TITLE
Taichi+Unity demo?

### DISCUSSION THREAD

**Discussion by bcolloran** [taichi-dev/taichi#5739] [Ideas]
On the ["why a new programming language"](https://docs.taichi-lang.org/docs/overview) page of the Taichi docs, there is a brief mention and screen capture of kernels written in Taichi being used in Unity. This sounds amazing--I've dabbled in Unity's compute shaders enough to know that it is incredibly expressive compared to the expressiveness and power of Tai chi. Are there any plans to support this functionality more fully in the future? In the meantime, would it be possible to share the code used to build that unity project? even if it's not fully supported yet it would be a very helpful demo (along the lines of the [AOT demo that you've already published ](https://github.com/taichi-dev/taichi-aot-demo)) just to start to think about what's possible and to get a feel for what that integration could look like.
Thank you!

**Comment by PENGUINLIONG** [comment]
The unity sample projects is available here: https://github.com/taichi-dev/Taichi-UnityExample It should run out of the box in the editor.

Although we only exchange data via `ComputeBuffer` for now, there is an on-going plan to support `RenderTexture` and `Texture*D`. It will be available real soon (#5520).  There is a lot of potential in Taichi to support all sorts procedural rendering in the future. ;)
